crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -775,6 +780,25 @@ impl CheckpointExecutor {
             )
             .filter_map(
                 |(txn, tx_digest, expected_fx_digest, effects, executed_fx_digest)| {
+                    let mut barrier_counts = BTreeMap::new();
```

Comment:
I think this can be problematic since it is still possible that we split a large checkpoint into multiple checkpoints.